main.go

```diff
@@ -732,9 +734,37 @@ func executeLinter(state *linterState, args []string) error {
 			state.Name)
 	}
 
-	processOutput(state, buf.Bytes())
+	cmdOutput, err := preprocessLinterPanic(state.Name, buf, err)
+	processOutput(state, cmdOutput)
 	debug("%s linter took %s", state.Name, time.Since(start))
-	return nil
+	return err
+}
+
+const panicExitStatus = 2
+
+func preprocessLinterPanic(name string, buf *bytes.Buffer, processError error) ([]byte, error) {
+	if exitStatus(processError) == panicExitStatus {
```

Comment:
I considered ignoring the exit status entirely. This is a small optimization, but maybe it's wrong. I'll wait to see what @alecthomas thinks. I'm fine removing the exit status check.

Regarding the error. The error returned here is printed as a warning by the acller. I think it's best to have all the warnings together at the end, so I think it's best to keep returning the error.